pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -166,6 +189,198 @@ func TestReconcileClusterPool(t *testing.T) {
 			},
 			expectPoolVersionChanged: true,
 		},
+		{
+			name: "poolVersion changes with new Inventory",
+			existing: []runtime.Object{
+				initializedPoolBuilder.Build(testcp.WithInventory([]string{"test-cdc-1"})),
+			},
+			inventory:                true,
+			expectPoolVersionChanged: true,
+		},
+		{
+			name: "poolVersion doens't change with existing inventory when entry added",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithInventory([]string{"test-cdc-1", "test-cdc-2"})),
+			},
+			inventory:                true,
+			expectPoolVersionChanged: false,
+			expectedPoolVersion:      "06983eaafac7f695",
+		},
+		{
+			name: "poolVersion changes when inventory removed",
+			existing: []runtime.Object{
+				inventoryPoolBuilder.Build(testcp.WithInventory([]string{})),
+			},
+			inventory:                false,
+			expectPoolVersionChanged: true,
+			expectedPoolVersion:      "06983eaafac7f695",
```

Comment:
This is actually a problem with the test phrasing:
```
assert.Equal(
	t, test.expectPoolVersionChanged, currentPoolVersion != expectedPoolVersion,
	"expectPoolVersionChanged is %t\ninitial %q\nfinal   %q",
        test.expectPoolVersionChanged, expectedPoolVersion, currentPoolVersion)
```
This could be "fixed" by changing `expectedPoolVersion` to `expectedInitialPoolVersion`.